### ISSUE TITLE
Microservices - rabbitmq options

### ISSUE THREAD

**Issue by capaio** [nestjs/nest#5587]
## Is your feature request related to a problem? Please describe.
 if you use a rabbitmq password that has a slash it is impossible to connect:

```javascript
 const user = 'username'
  const pwd = 'nGasd!56J/-)Q6'
  const host = 'localhost';
  const port = 5672;

  const microservicePayment = app.connectMicroservice<MicroserviceOptions>({
    transport: Transport.RMQ,
    options: {
      urls: [`amqp://${user}:${pwd}@${host}:${port}`],
      queue: 'queuename',
      queueOptions: {
        durable: false
      },
    },
  });
```
you get this error:

```
{
"err": {
  "cause": {
     "errno": -3001,
      "code": "EAI_AGAIN",
       "syscall": "getaddrinfo",
      "hostname": "username"
    },
     "isOperational": true,
     "errno": -3001,
     "code": "EAI_AGAIN",
     "syscall": "getaddrinfo",
     "hostname": "username"
 }
}

```
## Describe the solution you'd like

At the moment the interface of rabbitmq option is:
```javascript
export interface RmqOptions {
    transport?: Transport.RMQ;
    options?: {
        urls?: string[];
        queue?: string;
        prefetchCount?: number;
        isGlobalPrefetchCount?: boolean;
        queueOptions?: any;
        socketOptions?: any;
        noAck?: boolean;
        serializer?: Serializer;
        deserializer?: Deserializer;
        replyQueue?: string;
    };
}
```

the option 'urls' is an array of strings. But it should be possible to use objects instead of string as described in 
https://www.squaremobius.net/amqp.node/channel_api.html#connecting-with-an-object-instead-of-a-url
```javascript
{
  protocol: 'amqp',
  hostname: 'localhost',
  port: 5672,
  username: 'guest',
  password: 'guest',
  locale: 'en_US',
  frameMax: 0,
  heartbeat: 0,
  vhost: '/',
}
```

it would be very useful to use this object too instead of just strings. LIke this:

```javascript
interface RmqUrl {
    protocol?: string;
    hostname?: string,
    port?: number,
    username?: string,
    password?: string,
    locale?: string,
    frameMax?: number,
    heartbeat?: number,
    vhost?: string,
}

export interface RmqOptions {
    transport?: Transport.RMQ;
    options?: {
        urls?: string[] | RmqUrl[];
        queue?: string;
        prefetchCount?: number;
        isGlobalPrefetchCount?: boolean;
        queueOptions?: any;
        socketOptions?: any;
        noAck?: boolean;
        serializer?: Serializer;
        deserializer?: Deserializer;
        replyQueue?: string;
    };
}
```

I have tried editing the code as I showed and it works proprerly

I edited the report several times to make it more clear, sorry if this made you confuse

**Comment by jmcdo29** [comment]
Would you like to create a PR to update the acceptable types?

**Comment by capaio** [comment]
Yes, It would be a pleasure!

**Comment by kamilmysliwiec** [comment]
Let us know if you need anything from us to proceed @capaio! 💪

**Comment by capaio** [comment]
I have opened a PR https://github.com/nestjs/nest/pull/5609

**Comment by kamilmysliwiec** [comment]
My fault, thanks @capaio! I haven't noticed it

**Comment by kamilmysliwiec** [comment]
Fixed in 7.5.0
